### PR TITLE
Add Auto Suggest API specification

### PR DESCRIPTION
**Pull request by jwood803**
Apologies for the WIP pull request, but I'm not exactly sure if I'm on the right track. From my understanding, this is needed to auto-generate the code for the Python Azure SDK.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by anuchandy** [comment]
@jwood803 thank you. Just trying to get some context here, saw that you are from Wintellect, are you in touch with anyone from Microsoft cognitive service team? if so can you can tag him/her in this PR so that service team also can take a look. In case you don't know anyone in cognitive team please let me know, i can find the contact and help you to progress this PR. Again thanks for the contribution!

**Comment by jwood803** [comment]
Thanks for getting back @anuchandy! I don't know anyone at Microsoft myself, but I'm sure some colleagues of mine do. I can ask them if that works.

Apologies for the noise in this PR. I kind of figured things out as they went. I'm sure I still have some updates here, too. 😄

**Comment by anuchandy** [comment]
@jwood803 no worries, i'm checking with cognitive service team to see whether they can take a look at this PR.

File: specification/cognitiveservices/data-plane/AutoSuggest/readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  add-credentials: true
+  payload-flattening-threshold: 2
+  namespace: azure.cognitiveservices.search.autosuggest
+  package-name: azure-cognitiveservices-search-autosuggest
+  clear-output-folder: true
+```
+``` yaml $(python) && $(python-mode) == 'update'
+python:
+  no-namespace-folders: true
+  output-folder: $(python-sdks-folder)/azure-cognitiveservices-search-newssearch/azure/cognitiveservices/search/newssearch
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/AutoSuggest/readme.md]: Please change "newssearch" for "autosuggest

File: specification/cognitiveservices/data-plane/AutoSuggest/readme.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```
+``` yaml $(python) && $(python-mode) == 'create'
+python:
+  basic-setup-py: true
+  output-folder: $(python-sdks-folder)/azure-cognitiveservices-search-newssearch/azure/cognitiveservices/search/newssearch
```
Comment (lmazuel) [inline specification/cognitiveservices/data-plane/AutoSuggest/readme.md]: Please change "newssearch" for "autosuggest

**Review by lmazuel — CHANGES_REQUESTED** [review]
Content LGTM, please fix the Python conf. Thanks!

**Comment by jwood803** [comment]
Thanks a ton for reviewing @lmazuel! I plan on doing the [sample](https://github.com/Azure-Samples/cognitive-services-python-sdk-samples), too, so if I notice anything missing I'll make appropriate updates.

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

**Review by lmazuel — APPROVED** [review]

**Comment by lmazuel** [comment]
@jwood803 Just to give you update, we need someone from the Autosuggest team to sign-off, but your PR looks good to me

**Comment by jwood803** [comment]
Thanks a ton, @lmazuel! I think I broke it when fixing the linting errors, but some extra guidance from that team would be very helpful. Thank you again!

**Comment by jwood803** [comment]
Looks like I've been slacking too much and this has already been implemented. Should this PR be closed?

**Comment by jwood803** [comment]
No worries, @lmazuel! I should have fixed the issues in this PR a while ago to help it ready for further review. The other PR looks better than mine, too 😄 

I would like to implement the samples in the [samples repository](https://github.com/Azure-Samples/cognitive-services-python-sdk-samples), but I'm not sure about the workflow before it becomes a Python package in PyPI. I'm guessing after [this PR](https://github.com/Azure/azure-sdk-for-python/pull/2710) gets merged the Python package gets auto-generated?

**Comment by lmazuel** [comment]
@jwood803 Yes, I need to configure the "setup.py" and do some testing, but indeed you got it :)

**Comment by jwood803** [comment]
Awesome, @lmazuel, thanks! I'll wait for the package to be generated and I'll hit up some samples! 😃 

I'll close this PR, but thank you for all the help! It's much appreciated!

**Pull request closed without merging**